File: src/main/java/com/github/bgomar/bgdevtoys/tools/ConsoleLoggerSettingsConfigurable.java
package com.github.bgomar.bgdevtoys.tools;

import com.github.bgomar.bgdevtoys.toolwindow.setup.PropertiesConsoleLoggerToolSetup;
import com.intellij.openapi.options.ConfigurableBase;
import org.jetbrains.annotations.NotNull;

public class ConsoleLoggerSettingsConfigurable extends ConfigurableBase<PropertiesConsoleLoggerToolSetup, ConsoleLoggerSettings> {

  public ConsoleLoggerSettingsConfigurable() {
    super("com.github.bgomar.consolelogger", "ConsoleLogger", "");
  }

  @Override
  protected @NotNull ConsoleLoggerSettings getSettings() {
    return ConsoleLoggerSettings.getInstance();
  }

  @Override
  protected PropertiesConsoleLoggerToolSetup createUi() {
    return new ConsoleLoggerTools.ConsoleLoggerConfig(getSettings());
  }
}

File: restfiles/settings/ConsoleLoggerSettingsConfigurable.kt
package com.github.bgomar.consolelogger.settings

import com.intellij.openapi.options.ConfigurableBase

class ConsoleLoggerSettingsConfigurable : ConfigurableBase<ConsoleLoggerConfigurableUI, ConsoleLoggerSettings>("com.github.bgomar.consolelogger", "ConsoleLogger", "") {

  override fun getSettings(): ConsoleLoggerSettings {
    return ConsoleLoggerSettings.instance
  }

  override fun createUi(): ConsoleLoggerConfigurableUI {
    return ConsoleLoggerConfigurableUI(settings)
  }
}
